hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tenant/OMTenantAssignUserAccessIdRequest.java

```diff
@@ -308,20 +302,20 @@ public OMClientResponse validateAndUpdateCache(
         principalInfo.addAccessId(accessId);
       }
       omMetadataManager.getPrincipalToAccessIdsTable().addCacheEntry(
-          new CacheKey<>(principal),
+          new CacheKey<>(userPrincipal),
           new CacheValue<>(Optional.of(principalInfo),
               transactionLogIndex));
 
       // Add to tenantGroupTable
-      // TODO: DOUBLE CHECK GROUP NAME USAGE
+      // TODO: TenantGroupTable is unused for now.
```

Comment:
Actually (mildly confusingly), `TenantGroupTable` is supposed to be used to store all Ranger **roles** created for this tenant (`tenant-UserRole` and `tenant-AdminRole`, at this point). -- Might be useful for the background sync thread.

The other one, `TenantRoleTable` is supposed to be used for storing the list of tenant roles (?). A concept @prashantpogde introduced in the design phase but I don't see how it would relevant at this point. Probably could remove it

Reply:
@prashantpogde do you have some insight on the final schema we should commit to with regards to these two tables and the background sync work?

Reply:
Discussed offline and filed HDDS-6396